src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -525,5 +525,14 @@ public static class Diagnostics {
 			defaultSeverity: DiagnosticSeverity.Error,
 			isEnabledByDefault: true
 		);
+
+		public static readonly DiagnosticDescriptor UnexpectedConditionalImmutability = new DiagnosticDescriptor(
+			id: "D2L0071",
+			title: "The [ConditionallyImmutable.OnlyIf] attribute cannot be applied to this member.",

```

Comment:
```suggestion
			title: "The [ConditionallyImmutable.OnlyIf] attribute cannot be used with methods.",
```

(more specific, so that it's easy to understand what's wrong without also looking at the line of code)

Reply:
Sounds good. I wasn't sure if there were other things it might be applied to in the future, which is why I left it generic.

Reply:
At this time the only things that can be generic are type decls and methods. If they add something else to the language we can adjust the diagnostic later :)